rero_ils/modules/items/api/circulation.py

```diff
@@ -849,15 +849,33 @@ def prior_checkout_actions(self, action_params):
             ):
                 item, receive_actions = self.receive(**action_params)
                 actions.update(receive_actions)
-            if loan['state'] == LoanState.ITEM_IN_TRANSIT_TO_HOUSE:
-                item, cancel_actions = self.cancel_loan(pid=loan.get('pid'))
+            elif loan['state'] == LoanState.ITEM_IN_TRANSIT_TO_HOUSE:
+                item, cancel_actions = self.cancel_loan(**action_params)
                 actions.update(cancel_actions)
                 del action_params['pid']
+                # TODO: Check what's wrong in this case because Loan is cancel
+                # but loan variable is not updated and after prior_checkout
+                # a checkout is done on the item (it becomes ON_LOAN)
         else:
             loan = get_loan_for_item(item_pid_to_object(self.pid))
             if (loan and loan['state'] != LoanState.ITEM_AT_DESK):
                 item, cancel_actions = self.cancel_loan(pid=loan.get('pid'))
                 actions.update(cancel_actions)
+        # CHECKOUT_1_2_2: checkout denied if some pending loan are linked to it
+        # with different patrons
+        # Except while coming from an ITEM_IN_TRANSIT_TO_HOUSE loan because
+        # the loan is cancelled and then came up in ON_SHELF to be checkout
+        # by the second patron.
+        if self.status == ItemStatus.ON_SHELF and \
+                loan['state'] != LoanState.ITEM_IN_TRANSIT_TO_HOUSE:
+            for res in self.get_item_loans_by_state(state=LoanState.PENDING):
+                if res.patron_pid != loan.get('patron_pid'):
+                    pending = Loan.get_record_by_pid(res.pid)
```

Comment:
since `res` is a `loan`, then you can remove the line 873:

`pending = Loan.get_record_by_pid(res.pid)`

and replace 
`pending.patron_pid` by `res.patron_pid` in line 876